components/server/ee/src/prebuilds/github-app.ts

```diff
@@ -192,7 +205,20 @@ export class GithubApp {
             const contextURL = `${repo.html_url}/tree/${branch}`;
             span.setTag('contextURL', contextURL);
 
-            let config = await this.prebuildManager.fetchConfig({ span }, user, contextURL);
+            const context = await this.contextParser.handle({ span }, user, contextURL) as CommitContext;
+            const config = await this.prebuildManager.fetchConfig({ span }, user, context);
+
+            // if it's a sub-repo of a multi-repo project, we look up the owner of the main repo
+            if (!!context.subRepositoryCheckoutInfo && (!project || context.repository.cloneUrl !== project.cloneUrl)) {
+                const owner = await this.findOwnerAndProject(installationId, context.repository.cloneUrl);
+                if (!owner) {
+                    log.info(`Did not find owner for cloneURL`, { cloneURL: context.repository.cloneUrl});
+                    return;
```

Comment:
What would the user see in this case? How would the system behave? Also, how would we observe this in the logs (metric instead?), i.e. is the log message actionable - if not, it should probably be a debug message?